teacher.rb

```diff
@@ -0,0 +1,15 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, age, name = 'Unknown', permission: true)
+    super(age, name, permission)
```

Comment:
Let's try to stay consistent with our naming, to avoid future issues😉. you are already making use of parent_permission, so try to stay consistent with it.
```suggestion
  def initialize(specialization, age, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission)
```